src/migration/1749656464598-SourceMemberFollowers.ts

```diff
@@ -0,0 +1,62 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class SourceMemberFollowers1749656464598 implements MigrationInterface {
+    name = 'SourceMemberFollowers1749656464598'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`
+            CREATE OR REPLACE FUNCTION increment_source_members_count()
+                RETURNS TRIGGER
+                LANGUAGE PLPGSQL
+                AS
+            $$
+            DECLARE
+                source_id_to_update TEXT;
+            BEGIN
+                IF TG_OP = 'DELETE' THEN
+                    source_id_to_update := OLD."sourceId";
+                ELSE
+                    source_id_to_update := NEW."sourceId";
+                END IF;
+
+                IF (TG_OP = 'DELETE' AND OLD.type = 'source') OR 
+                   (TG_OP != 'DELETE' AND NEW.type = 'source') THEN
+                    
+                    UPDATE source
+                    SET flags = jsonb_set(
+                                  flags,
+                                  '{totalMembers}',
+                                  to_jsonb((
+                                      SELECT COUNT(DISTINCT "userId")
+                                      FROM content_preference
+                                      WHERE "sourceId" = source_id_to_update
+                                      AND type = 'source'
+                                      AND status IN ('follow', 'subscribed')
```

Comment:
Might need an index here

```sql
Aggregate  (cost=7426.33..7426.34 rows=1 width=8) (actual time=3257.892..3257.894 rows=1 loops=1)
  ->  Sort  (cost=7426.22..7426.28 rows=22 width=21) (actual time=3255.881..3256.482 rows=13507 loops=1)
        Sort Key: ""userId""
        Sort Method: quicksort  Memory: 385kB
        ->  Index Scan using ""IDX_content_preference_source_id"" on content_preference  (cost=0.57..7425.73 rows=22 width=21) (actual time=3.467..3208.035 rows=13507 loops=1)
              Index Cond: (""sourceId"" = 'hn'::text)
              Filter: ((status = ANY ('{follow,subscribed}'::text[])) AND (type = 'source'::text))
              Rows Removed by Filter: 668
Planning Time: 0.372 ms
Execution Time: 3257.966 ms
```

Reply:
Something like this might work

```sql
CREATE INDEX CONCURRENTLY IF NOT EXISTS "IDX_content_preference_sourceid_type_status_userid"
ON public.content_preference USING btree ("sourceId", "type", "status", "userId");
```

Reply:
Can't use `CONCURRENTLY` in migrations because it runs in transaction

Reply:
Either way this should be added manually because of table size.